common/core/desktop/src/km_kbp_context_api.cpp

```diff
@@ -76,6 +76,11 @@ namespace {
 
       *sz_ptr = buf_size - (e - i);
 
+      // Skip over any final markers - they are execluded from context
+      while(ci->type == KM_KBP_CT_MARKER) {
+        ci++;
+      }
```

Comment:
This was a simple bug that would cause an assertion failure if the last item(s) in the context was (were) markers, as then we'd never get to the `KM_KBP_CT_END` item.

Reply:
Yep. In the way this function is intended to be used, we call first to get buf size, which returns the size without markers. So that means we'd always trigger this issue if there were markes at the end of the context!